codex/codex.nim

```diff
@@ -113,12 +113,10 @@ proc start*(s: CodexServer) {.async.} =
   notice "Starting codex node"
 
   await s.repoStore.start()
-  s.restServer.start()
-
-  s.codexNode.contracts = await bootstrapInteractions(s.config, s.repoStore)
-  await s.codexNode.start()
   s.maintenance.start()
 
+  await s.codexNode.switch.start()
```

Comment:
Node-start has to be after the marketplace interaction bootstrap, because the bootstrap sets 3 interaction objects that the node-start will start if they are set.
Yeah it's a bit of a tangle.

Reply:
A proposal for cleaning up the application component lifecycle is here: https://github.com/codex-storage/nim-codex/issues/544